_posts/2019-07-19-paho-mqtt.md

```diff
@@ -0,0 +1,114 @@
+ 
+Navigate to the directory where you have cloned our 'examples' repository:
+
+```
+$ cd examples
+$ cd mosquitto-server
+```
+
+Alter the ACL file `permissions.acl` with your WoTT device details for the relevant permissions (remember we want to grant some access to your client device). To find the WoTT Device ID:
+
+```
+$ sudo wott-agent whoami
+```
+
+We are still using the topic `/wott/temperature` to publish and subscribe messages to and from.
+
+## Setting up the client (Paho)
+
+This is where our instructions differ from setting up the Mosquitto client. It is not significantly more complicated but Paho has a different set up. First, you will need to have Paho on your system.
+Navigate to the directory containing our `examples` respository. Navigate to the `paho-mqtt-client` folder like so:
+
+```
+$ cd examples
+$ cd paho-mqtt-client
+$ pip3 install -r requirements.txt
+```
+
+There are two files in this folder - a `requirements.txt` file and the client itself `paho-connect-wott.py`. The above command installs the version of the Paho client needed for this example. 
+
+**Note:** you can use whatever version of the `pip` tool you have.
+
+Now that Paho is installed, open the `paho-connect-wott.py` file in your favourite text editor. Towards the end of the document (line 33) you will notice this line:
+
+```
+client.connect('ServerDeviceID.d.wott.local', 8883, 60)
+```
+Substitute your Server device's WoTT ID into here in order to connect to the server. Save and exit. 
+
+Feel free to try and run the client now using `sudo python3 paho-connect-wott.py`. You will notice that this fails. 
+This is because we haven't established a route for the client to connect to the server device through - we need to alter the `/etc/hosts` file on the client to add a route to the server. In the Mosquitto example, we set this up using Docker, but here you will need to manually add the hostname. So, in whatever text editor you prefer (for the purposes of demonstration I am using `gedit`):
+
+```
+$ sudo gedit /etc/hosts
+```
+
+You will see a file containing something like this:
+
+```
+127.0.0.1    localhost
+127.0.1.1    your_machine's_name
+[...]
+
+# The following lines are desirable for IPv6 capable hosts
+[...]
+```
+Add your server device's IP followed by the 'domain' that you will be calling it by- the device's WoTT ID.
+
+```
+IP.Of.Docker.Container  ServerDeviceID.d.local
+```
+
+Save and exit. Now you have a route to the server!
+
+**Note:** you *must* use the IP of the Docker container running the server process (if you are using our Mosquitto server). To get this run `ip addr show` or `ifconfig` (or whatever command you are familiar with). You will need the `inet` address for Docker specifically - usually named something like `docker0`.
```

Comment:
No this is wrong. It should be the IP on the network. The IP of `docker0` will only work locally,  since it's a internal IP (localhost). Technically speaking, the reason why this works is because we listen on `0.0.0.0` implicitly when we start the docker container, which means that we listen on all local addresses.

Reply:
That's strange - I couldn't use my normal IP, it would break. So I had to use the IP for the Docker container

Reply:
Could be related to your firewall or something.